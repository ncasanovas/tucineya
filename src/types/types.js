export const types = {
  login: "[auth] login",
  admin: "[auth] admin",
  logout: "[auth] logout",
};
