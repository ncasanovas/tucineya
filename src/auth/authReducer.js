import { types } from "../types/types";

export const authReducer = (state = {}, action) => {
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true,
        admin: false,
      };
    case types.logout:
      return {
        logged: false,
      };
    case types.admin:
      return {
        ...action.payload,
        logged: true,
        admin: true,
      };

    default:
      return state;
  }
};
