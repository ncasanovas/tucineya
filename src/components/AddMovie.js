import React, { useState } from "react";
import axios from "axios";

import { Modal, Button } from "react-bootstrap";
import { useHistory } from "react-router";

export const AddMovie = () => {
  const [show, setShow] = useState(false);

  const history = useHistory();

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <div className="mt-2">
      <Button variant={"primary"} onClick={handleShow}>
        Agregar Pelicula
      </Button>

      <Modal
        show={show}
        onHide={handleClose}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>Pelicula</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          ¿Está seguro de querer eliminar los usuarios seleccionados?
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary btn-secondary" onClick={handleClose}>
            Volver
          </Button>
          <Button variant="danger">Eliminar</Button>
        </Modal.Footer>
      </Modal>
    </div>
  );
};
