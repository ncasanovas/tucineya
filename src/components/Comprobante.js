import React from "react";
import { useHistory, HashRouter, Link } from "react-router-dom";

export const Comprobante = () => {
  const printReceipt = () => {
    var prtContent = document.getElementById("tarjeta");
    var WinPrint = window.open(
      "",
      "",
      "left=200,top=200,width=800,height=900,toolbar=0,scrollbars=0,status=0"
    );
    WinPrint.document.write(prtContent.innerHTML);
    WinPrint.document.close();
    WinPrint.focus();
    WinPrint.print();
    WinPrint.close();
  };

  const printReceipt2 = () => {
    var prtContent = document.getElementById("tarjeta2");
    var WinPrint = window.open(
      "",
      "",
      "left=0,top=0,width=800,height=900,toolbar=0,scrollbars=0,status=0"
    );
    WinPrint.document.write(prtContent.innerHTML);
    WinPrint.document.close();
    WinPrint.focus();
    WinPrint.print();
    WinPrint.close();
  };

  const { idNombrePelicula, idNumeroSala, fechaYhora, nombre } = JSON.parse(
    sessionStorage.getItem("Pelicula")
  );

  return (
    <div className="App">
      <header className="App-header">
        <div className="card">
          <HashRouter>
            <Link className="col-1" to="/buscarPelicula">
              Atras
            </Link>
          </HashRouter>
          <div className="card-body" id="tarjeta2" z-index="3200">
            <h5 className="card-title">Comprobante</h5>

            <div>
              <h3>{idNombrePelicula[0]}</h3>
              <p>Sala numero:{idNumeroSala}</p>
              <p>Precio: {sessionStorage.getItem("precioTotal")}</p>
              <p>Cine: {nombre}</p>
              <p>
                Butacas elegidas:{" "}
                {JSON.stringify(sessionStorage.getItem("butacasElegidas"))}
              </p>
              <p>Fecha: {fechaYhora}</p>

              <h1>Tu Cine Ya</h1>
            </div>
          </div>
        </div>

        <button
          className="hide-on-print btn btn-primary"
          onClick={printReceipt2}
        >
          Imprimir
        </button>
      </header>
    </div>
  );
};

export default Comprobante;
