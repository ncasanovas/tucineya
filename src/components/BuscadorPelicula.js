import React, { useState, useContext, useEffect } from "react";
import axios from "axios";
import { useHistory } from "react-router-dom";

import { AuthContext } from "../auth/AuthContext";
import { MovieContext } from "./MovieContext";

import { types } from "../types/types";
import { ResultadoPeliculas } from "./ResultadoPeliculas";
import { ElegirCine } from "./ElegirCine";

export const BuscadorPelicula = () => {
  const [inputValue, setInputValue] = useState();
  const [user, setUser] = useState();
  const [localidades, setLocalidades] = useState();
  const { dispatch } = useContext(AuthContext);
  const { movies, setMovies } = useContext(MovieContext);
  const history = useHistory();

  const onChangeinput = (e) => {
    setInputValue(e.target.value);
  };

  useEffect(async () => {
    let user = localStorage.getItem("usuario");
    setUser(user);
    await axios
      //.get("http://localhost:4000/api/cines/")
      .get("https://tucineya.herokuapp.com/api/cines/")
      .then((res) => {
        setLocalidades(res.data[0]);
      });
  }, [movies]);

  const onClickSearchMovie = async (e) => {
    e.target.reset();
    e.preventDefault();
    await axios
      //.get(`http://localhost:4000/api/movies/${inputValue}`)
      .get(`https://tucineya.herokuapp.com/api/movies/${inputValue}`)
      .then((res) => {
        setMovies(res.data.data[0]);
      })
      .catch((e) => {
        console.log(e);
      });
    setInputValue("");
  };

  const handleLogout = () => {
    console.log("Click");
    dispatch({
      type: types.logout,
    });
    history.replace("./");
  };

  return (
    <div className="row mt-4">
      <div className="d-flex justify-content-end">
        <h4 style={{ color: "white" }}>Bienvenido {user}!</h4>
        <button
          className="nav-item nav-link btn white-text"
          onClick={() => handleLogout}
        >
          Cerrar Sesi√≥n
        </button>
      </div>
      <div className="col mb-3">
        <form className="mb-4" onSubmit={() => onClickSearchMovie()}>
          <input
            type="text"
            placeholder="Busque por titulo"
            onChange={onChangeinput}
          />
          <button className="btn-primary">Buscar</button>
        </form>
        <div id="IdDropdown">
          <ElegirCine localidades={localidades} />
        </div>
      </div>
      <div className="col">{movies ? <ResultadoPeliculas /> : null}</div>
    </div>
  );
};
