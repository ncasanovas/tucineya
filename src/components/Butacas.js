import React, { useEffect, useState, useContext } from "react";
import axios from "axios";
import { useHistory, HashRouter, Link } from "react-router-dom";

import { AuthContext } from "../auth/AuthContext";
import { types } from "../types/types";
import { MovieContext } from "./MovieContext";

export const Butacas = () => {
  const asientosFila = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"];
  const filas = ["1", "2", "3", "4", "5", "6", "7"];
  const [butacaElegida, setButacaElegida] = useState([]);
  const [butacasConfirmadas, setButacasConfirmadas] = useState([]);
  const [confirmar, setConfirmar] = useState(false);
  const { dispatch } = useContext(AuthContext);
  const history = useHistory();
  const { movies, idSala } = useContext(MovieContext);

  useEffect(async () => {
    if (idSala !== null) {
      await axios
        //.get(`http://localhost:4000/api/butacas/${idSala}`)
        .get(`https://tucineya.herokuapp.com/api/butacas/${idSala}`)
        .then((res) => {
          if (res.data !== null) {
            if (
              res.data[0][0].butacas === "" ||
              res.data[0][0].butacas === null
            ) {
              localStorage.setItem("butacas", "");
              setButacasConfirmadas(localStorage.getItem("butacas"));
            } else {
              let butacas = res.data[0][0].butacas.split(",");
              localStorage.setItem("butacas", butacas);
              setButacasConfirmadas(butacas);
            }
          } else {
            setButacasConfirmadas(localStorage.getItem("butacas"));
          }
        });
    }
  }, [confirmar]);

  const seleccionarButaca = (fila, asiento) => {
    if (butacaElegida.includes(fila + asiento)) {
      // Si el asiento ya está seleccionado en el array
      let index = butacaElegida.indexOf(fila + asiento); // Guardo el indice en donde esté la butaca
      butacaElegida.splice(index, 1); // Y la borro del array segun el indice que encontró, el segundo argumento es para que borre solo ese registro
      setButacaElegida([...butacaElegida]); // setea las butacas con el array y ya sin la que borramos
    } else {
      setButacaElegida([...butacaElegida, fila + asiento]); // setea las butacas seleccionadas si no está en el array de butacaElegida
    }
  };

  const confirmarSeleccion = async () => {
    if (butacaElegida.length !== 0) {
      await axios
        .post("https://tucineya.herokuapp.com/api/butacas", {
          //.post("http://localhost:4000/api/butacas", {
          butacas: butacaElegida,
          idSala: idSala,
        })
        .then(() => {
          setButacaElegida([]);
          setConfirmar(!confirmar);
        });
    }
  };

  const handleLogout = () => {
    dispatch({
      type: types.logout,
    });
    history.replace("./");
  };

  return (
    <div className="row">
      <div className="d-flex justify-content-end">
        <HashRouter>
          <Link
            to="/buscarPelicula"
            className="col-2 align-items-center"
            onClick={() => {
              localStorage.removeItem("butacas");
            }}
          >
            Atras
          </Link>
        </HashRouter>
        <button
          className="nav-item nav-link btn white-text"
          onClick={handleLogout}
        >
          Cerrar Sesión
        </button>
      </div>
      <div className="col">
        {movies !== [] ? null : (
          <img
            src={movies[0].posterPelicula}
            alt={movies[0].idNombrePelicula}
          />
        )}
      </div>
      <div className="col">
        <ul className="showcase">
          <li>
            <div className="seat"></div>
            <small>Disponible</small>
          </li>
          <li>
            <div className="seat selected"></div>
            <small>Seleccionado</small>
          </li>
          <li>
            <div className="seat occupied"></div>
            <small>Ocupado</small>
          </li>
        </ul>

        <div className="container ">
          <div className="screen"></div>
          <div className="row">
            {filas.map((fila, j) => {
              return (
                <div className="row" key={j} value={fila}>
                  {asientosFila.map((asiento, i) => {
                    return (
                      <div
                        className={
                          butacasConfirmadas.includes(fila + asiento)
                            ? "seat occupied"
                            : butacaElegida.includes(fila + asiento)
                            ? "seat selected"
                            : "seat"
                        }
                        style={
                          butacasConfirmadas.includes(fila + asiento)
                            ? { pointerEvents: "none" }
                            : null
                        }
                        disabled
                        unselectable="true"
                        key={i}
                        value={asiento}
                        onClick={() => seleccionarButaca(fila, asiento)}
                      ></div>
                    );
                  })}
                </div>
              );
            })}
          </div>
          <button
            className="btn btn-primary"
            onClick={() => confirmarSeleccion()}
          >
            Confirmar
          </button>
        </div>

        <p className="text">
          You have selected <span id="count">0</span> seats for the price of $
          <span id="total">0</span>!
        </p>
      </div>
    </div>
  );
};
