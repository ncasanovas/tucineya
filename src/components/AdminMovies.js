import React, { useContext } from "react";
import { Card, Button } from "react-bootstrap";
import { AddMovie } from "./AddMovie";
import { MoviesContext } from "./MoviesContext";

export const AdminMovies = () => {
  const { moviesContext } = useContext(MoviesContext);

  return (
    <div className="p-0 m-0">
      <div>
        {moviesContext ? (
          <div className="overflow-auto d-flex">
            {moviesContext.map((movie, i) => {
              return (
                <div className="col-4 overflow-auto d-flex" key={i}>
                  <Card>
                    <Card.Img
                      className="img-fluid"
                      id="imgPelicula"
                      variant="top"
                      src={`https://${movie.posterPelicula}`}
                    />
                    <Card.Body>
                      <Card.Title id="cardText">
                        {movie.nombrePelicula}
                      </Card.Title>
                      <Card.Text>{movie.sinopsis}</Card.Text>
                    </Card.Body>
                  </Card>
                </div>
              );
            })}
          </div>
        ) : (
          <div className="d-flex justify-content-center">
            <div
              className="spinner-border text-primary d-flex justify-content-center"
              role="status"
            >
              <span className="visually-hidden ">Loading...</span>
            </div>
          </div>
        )}
      </div>
      {moviesContext ? (
        <AddMovie />
      ) : //<Button variant="primary">Agregar Pelicula</Button>
      null}
    </div>
  );
};
