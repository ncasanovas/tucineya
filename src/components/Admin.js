import React, { useState, useEffect, useContext } from "react";
import axios from "axios";
import { HashRouter, Link } from "react-router-dom";

import { AdminModal } from "./AdminModal";
import { AdminMovies } from "./AdminMovies";
import { MoviesContext } from "./MoviesContext";

export const Admin = () => {
  const [users, setUsers] = useState();
  const [deleteUsers, setDeleteUsers] = useState([]);

  const { setMoviesContext } = useContext(MoviesContext);

  useEffect(() => {
    const data = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/users/")
        .get("http://localhost:4000/api/users/")
        .then((res) => {
          setUsers(res.data[0]);
        });
    };
    const datamovies = async () => {
      await axios
        //.get("https://tucineya.herokuapp.com/api/movies/")
        .get("http://localhost:4000/api/movies/")
        .then((res) => {
          setMoviesContext(res.data[0]);
        });
    };
    data();
    datamovies();
  }, []);

  const onChangeCheck = (e) => {
    if (e.target.checked) {
      setDeleteUsers([...deleteUsers, e.target.value]);
    } else {
      for (let i = 0; i < deleteUsers.length; i++) {
        if (deleteUsers[i] === e.target.value) {
          deleteUsers.splice(i, 1);
          setDeleteUsers([...deleteUsers]);
        }
      }
    }
  };

  return (
    <div id="admin" className="container-fluid row">
      <div className="d-flex justify-content-end">
        <HashRouter>
          <Link to="/" className="col-2">
            Atras
          </Link>
        </HashRouter>
      </div>
      <div className="col-3 col-sm-4 col-md-3 col-lg-3">
        <h4 className="mb-4">Lista de usuarios</h4>
        <div className="row">
          {users ? (
            <ul className="list-group text-start col">
              {users.map((user, i) => {
                return (
                  <li className="list-group-item" key={i}>
                    <input
                      className="form-check-input me-1"
                      type="checkbox"
                      value={user.mail}
                      aria-label="..."
                      onChange={onChangeCheck}
                    />
                    {user.nombre} {user.apellido}
                  </li>
                );
              })}
            </ul>
          ) : (
            <div className="spinner-border text-primary m-6" role="status">
              <span className="visually-hidden ">Loading...</span>
            </div>
          )}
        </div>
        <div>
          {users ? (
            <div>
              <AdminModal
                deleteUsers={deleteUsers}
                setDeleteUsers={setDeleteUsers}
              />
            </div>
          ) : null}
        </div>
      </div>
      <div className="col-sm-12 col-md-9 col-lg-9">
        <h4 className="mb-4">Peliculas</h4>
        <AdminMovies />
      </div>
    </div>
  );
};

export default Admin;
