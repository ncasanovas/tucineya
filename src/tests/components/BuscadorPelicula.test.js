import React from "react";
import { mount, shallow } from "enzyme";

import { MovieContext } from "../../components/MovieContext";
import { BuscadorPelicula } from "../../components/BuscadorPelicula";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";
import axios from "axios";

describe("Tests en componente BuscadorPelicula.js", () => {
  const movieTest = {
    nombrePelicula: "MovieTest",
    posterPelicula: "http://demo.test.com",
    sinopsis: "Pelicula de prueba",
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
    },
  };

  //Se hacen 2 providers por tener en uno para el dispatch del login y el otro para manejar la pelicula buscada
  let wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MovieContext.Provider value={movieTest}>
        <BuscadorPelicula />
      </MovieContext.Provider>
    </AuthContext.Provider>
  );

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar los cines de la base de datos ", async () => {
    // Hace la peticion a la base de datos y esperaría que no regrese un resultado vacío
    jest.setTimeout(5000);
    await axios.get("https://tucineya.herokuapp.com/api/cines/").then((res) => {
      expect(res.data).not.toEqual([[]]);
      expect(typeof res.data.length).toBe("number");
    });
    expect(wrapper.find("#IdDropdown").update()).toHaveLength(1);
  });

  test("Debe mostrar las peliculas de la base de datos, según lo que busque el usuario", async () => {
    // Hace la peticion a la base de datos y esperaría que no regrese un resultado vacío
    const inputValue = "Kong";
    jest.setTimeout(5000);

    await axios
      .get(`https://tucineya.herokuapp.com/api/movies/${inputValue}`)
      .then((res) => {
        expect(res.data).not.toEqual([[]]);
      });
  });
});
