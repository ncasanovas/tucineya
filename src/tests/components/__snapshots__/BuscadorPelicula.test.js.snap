// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en componente BuscadorPelicula.js Debe mostrarse correctamente 1`] = `
<div
  className="row mt-4"
>
  <div
    className="d-flex justify-content-end"
  >
    <h4
      style={
        Object {
          "color": "white",
        }
      }
    >
      Bienvenido 
      !
    </h4>
    <button
      className="nav-item nav-link btn white-text"
      onClick={[Function]}
    >
      Cerrar Sesión
    </button>
  </div>
  <div
    className="col mb-3"
  >
    <form
      className="mb-4"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Busque por titulo"
        type="text"
      />
      <button
        className="btn-primary"
      >
        Buscar
      </button>
    </form>
    <div
      id="IdDropdown"
    >
      <div
        className="dropdown"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-primary"
          disabled={false}
          id="dropdown-basic"
          onClick={[Function]}
          type="button"
        >
          Localidades
        </button>
        <div>
          <ul
            className="list-group list-group-flush"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col"
  />
</div>
`;

exports[`Tests en componente ResultadoPelicula.js Debe mostrarse correctamente 1`] = `
<div
  className="row mt-4"
>
  <div
    className="d-flex justify-content-end"
  >
    <h4
      style={
        Object {
          "color": "white",
        }
      }
    >
      Bienvenido 
      !
    </h4>
    <button
      className="nav-item nav-link btn white-text"
      onClick={[Function]}
    >
      Cerrar Sesión
    </button>
  </div>
  <div
    className="col mb-3"
  >
    <form
      className="mb-4"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Busque por titulo"
        type="text"
      />
      <button
        className="btn-primary"
      >
        Buscar
      </button>
    </form>
    <div
      id="IdDropdown"
    >
      <div
        className="dropdown"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-primary"
          disabled={false}
          id="dropdown-basic"
          onClick={[Function]}
          type="button"
        >
          Localidades
        </button>
        <div>
          <ul
            className="list-group list-group-flush"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col"
  />
</div>
`;
